### PR TITLE
FIX #1223: Parameter containing unicode chars fail to serialize (MS W…

### LINKED ISSUE #1223: Parameter containing unicode chars fail to serialize (MS Windows)
## Issue
A parameter set to contain unicode characters will make luigi fail with a UnicodeEncodeError on Windows 7.

## Root cause
Luigi serializes parameters by using the `str` function instead of `unicode`.

## Fix
In file `luigi/parameter.py`, class `Parameter`, function `serialize`, replace:

	> return str(x)
	
With:

	> return unicode(x)
	
## Steps to reproduce

Create a file named "test.py" filled with:

	> # -*- coding: iso-8859-1 -*-
	> 
	> import luigi
	> 
	> class UnicodeTask (luigi.WrapperTask):
	> 	unicode_param = luigi.Parameter(u"é")
	> 	
	> if __name__ == '__main__':
	>     luigi.run(main_task_cls=UnicodeTask)
	
Open the Windows command line (CMD.EXE) and run:

	> python test.py --local-scheduler
	
Luigi crashes with a UnicodeEncodeError:

	> Traceback (most recent call last):
	>   File "test.py", line 9, in <module>
	>     luigi.run(main_task_cls=UnicodeTask)
	>   File "C:\Users\_\Documents\TEST\luigi-bug\luigi\interface.py", line 421, in run
	>     tasks = interface.parse(cmdline_args, main_task_cls=main_task_cls)
	>   File "C:\Users\_\Documents\TEST\luigi-bug\luigi\interface.py", line 302, in parse
	>     return self.parse_task(cmdline_args, main_task_cls)
	>   File "C:\Users\_\Documents\TEST\luigi-bug\luigi\interface.py", line 299, in parse_task
	>     return [task_cls(**task_params)]
	>   File "C:\Users\_\Documents\TEST\luigi-bug\luigi\task_register.py", line 97, in __call__
	>     h[k] = instantiate()
	>   File "C:\Users\_\Documents\TEST\luigi-bug\luigi\task_register.py", line 78, in instantiate
	>     return super(Register, cls).__call__(*args, **kwargs)
	>   File "C:\Users\_\Documents\TEST\luigi-bug\luigi\task.py", line 265, in __init__
	>     task_id_parts.append('%s=%s' % (param_name, param_objs[param_name].serialize(param_value)))
	>   File "C:\Users\_\Documents\TEST\luigi-bug\luigi\parameter.py", line 276, in serialize
	>     return str(x)
	> UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 0: ordinal not in range(128)

### PR DESCRIPTION
**Pull request by ThQ**
…indows)

### CODE REVIEW

**Comment by erikbern** [comment]
great if you can add a unit test

**Comment by Tarrasch** [comment]
Also, we would like to see the commits all squashed into one.

**Comment by toidi** [comment]
Serialization must return byte array is't it?

**Comment by Tarrasch** [comment]
@ThQ. Can you amend the 5 commits into a single commit and then ping me afterwards? Otherwise it's ready to be merged.

File: test/unicode_parameter_test.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from helpers import unittest
+
+import luigi
+
+
+class UnicodeParameterTest(unittest.TestCase):
```
Comment (Tarrasch) [inline test/unicode_parameter_test.py]: Actually you shouldn't need to create a whole new file for this I think. Can't you just delete this file and move this test to `parameter_test.py`?

**Comment by Tarrasch** [comment]
Closing due to inactivity. Feel free to reopen when you have time.

**Pull request closed without merging**